frontend/catalyst/compiler.py

```diff
@@ -174,12 +174,13 @@ def run_writing_command(command: List[str], compile_options: Optional[CompileOpt
         "quantum-bufferize",
         "func-bufferize",
         "func.func(finalizing-bufferize)",
+        "canonicalize",
         "func.func(buffer-hoisting)",
         "func.func(buffer-loop-hoisting)",
         "func.func(buffer-deallocation)",
         "convert-arraylist-to-memref",
-        "canonicalize",
         "convert-bufferization-to-memref",
+        "canonicalize",
```

Comment:
```suggestion
        "canonicalize", # Must be after convert-bufferization-to-memref
                                # otherwise there are issues in lowering of dynamic tensors.
```